src/main.rs

```diff
@@ -74,10 +75,21 @@ fn main() -> Result<i32> {
         true => Policy::limited(args.max_redirects.unwrap_or(10)),
         false => Policy::none(),
     };
+    let timeout = if let Some(timeout) = args.timeout {
+        let timeout = timeout.as_duration();
+        if timeout == Duration::default() {
+            None
+        } else {
+            Some(timeout)
+        }
+    } else {
+        None
+    };
```

Comment:
Maybe this logic could go inside `as_duration()`, and that could return an `Option<Duration>` instead?